crates/red_knot_python_semantic/src/semantic_index/re_exports.rs

```diff
@@ -29,39 +29,57 @@ use crate::{module_name::ModuleName, resolve_module, Db};
 #[salsa::tracked(return_ref)]
 pub(super) fn exported_names(db: &dyn Db, file: File) -> FxHashSet<Name> {
     let module = parsed_module(db.upcast(), file);
-    let mut finder = ExportFinder::new(db, file);
+    let mut exports = FxHashSet::default();
+    let mut finder = ExportFinder::new(db, file, ScopeKind::Global, &mut exports);
     finder.visit_body(module.suite());
-    finder.exports
+    exports
```

Comment:
Nit: I prefer to store the owned hash set on the finder and return it from there. It results in fewer lifetimes
```suggestion
    finder.exports
```

Reply:
Okay, I think it makes sense now, seeing how you use it as a nested visitor. You could use `std::mem::take` to take the exports and then assign it to the inner visitor but not sure if that's worth getting rid of the lifetime.

Reply:
Nice, the suggestion of using `std::mem::take()` made things a lot cleaner!